### ISSUE TITLE
Please support PHP 8.2

### ISSUE THREAD

**Issue by datenleck** [owncloud/music#1056]
Since NC 26 ist supporting PHP 8.2 the apps should do, too. 
Currently I cannot install the very nice app because of PHP 8.2.

**Comment by jjasoncool** [comment]
additional log in NC 26:
```log
Error: Class "OCP\AppFramework\Db\Mapper" not found in /var/www/html/custom_apps/music/lib/Db/BaseMapper.php:29
Stack trace:
#0 /var/www/html/lib/composer/composer/ClassLoader.php(578): include()
#1 /var/www/html/lib/composer/composer/ClassLoader.php(432): Composer\Autoload\ClassLoader::Composer\Autoload\{closure}('/var/www/html/c...')
#2 /var/www/html/custom_apps/music/lib/Db/ArtistMapper.php(24): Composer\Autoload\ClassLoader->loadClass('OCA\\Music\\Db\\Ba...')
#3 /var/www/html/lib/composer/composer/ClassLoader.php(578): include('/var/www/html/c...')
#4 /var/www/html/lib/composer/composer/ClassLoader.php(432): Composer\Autoload\ClassLoader::Composer\Autoload\{closure}('/var/www/html/c...')
#5 /var/www/html/custom_apps/music/lib/App/Music.php(363): Composer\Autoload\ClassLoader->loadClass('OCA\\Music\\Db\\Ar...')
#6 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(171): OCA\Music\App\Music->OCA\Music\App\{closure}(Object(OC\AppFramework\DependencyInjection\DIContainer))
#7 /var/www/html/3rdparty/pimple/pimple/src/Pimple/Container.php(122): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(Pimple\Container))
#8 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(138): Pimple\Container->offsetGet('ArtistMapper')
#9 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(487): OC\AppFramework\Utility\SimpleContainer->query('ArtistMapper')
#10 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(465): OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback('ArtistMapper')
#11 /var/www/html/custom_apps/music/lib/App/Music.php(272): OC\AppFramework\DependencyInjection\DIContainer->query('ArtistMapper')
#12 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(171): OCA\Music\App\Music->OCA\Music\App\{closure}(Object(OC\AppFramework\DependencyInjection\DIContainer))
#13 /var/www/html/3rdparty/pimple/pimple/src/Pimple/Container.php(122): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(Pimple\Container))
#14 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(138): Pimple\Container->offsetGet('ArtistBusinessL...')
#15 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(487): OC\AppFramework\Utility\SimpleContainer->query('ArtistBusinessL...')
#16 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(465): OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback('ArtistBusinessL...')
#17 /var/www/html/custom_apps/music/lib/App/Music.php(576): OC\AppFramework\DependencyInjection\DIContainer->query('ArtistBusinessL...')
#18 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(171): OCA\Music\App\Music->OCA\Music\App\{closure}(Object(OC\AppFramework\DependencyInjection\DIContainer))
#19 /var/www/html/3rdparty/pimple/pimple/src/Pimple/Container.php(122): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(Pimple\Container))
#20 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(138): Pimple\Container->offsetGet('Scanner')
#21 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(487): OC\AppFramework\Utility\SimpleContainer->query('Scanner')
#22 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(465): OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback('Scanner')
#23 /var/www/html/custom_apps/music/lib/Hooks/FileHooks.php(75): OC\AppFramework\DependencyInjection\DIContainer->query('Scanner')
#24 /var/www/html/custom_apps/music/lib/Hooks/FileHooks.php(62): OCA\Music\Hooks\FileHooks::handleUpdated(Object(OC\Files\Node\File), Object(OC\AppFramework\DependencyInjection\DIContainer))
#25 [internal function]: OCA\Music\Hooks\FileHooks::updated(Object(OC\Files\Node\File))
#26 /var/www/html/lib/private/Hooks/EmitterTrait.php(105): call_user_func_array(Array, Array)
#27 /var/www/html/lib/private/Hooks/PublicEmitter.php(40): OC\Hooks\BasicEmitter->emit('\\OC\\Files', 'postWrite', Array)
#28 /var/www/html/lib/private/Files/Node/Root.php(143): OC\Hooks\PublicEmitter->emit('\\OC\\Files', 'postWrite', Array)
#29 /var/www/html/lib/private/Files/Node/Node.php(129): OC\Files\Node\Root->emit('\\OC\\Files', 'postWrite', Array)
#30 /var/www/html/lib/private/Files/Node/Folder.php(200): OC\Files\Node\Node->sendHooks(Array, Array)
#31 /var/www/html/lib/private/Files/SimpleFS/NewSimpleFile.php(121): OC\Files\Node\Folder->newFile('cities1000.csv', Resource id #1165)
#32 /var/www/html/lib/private/Files/SimpleFS/NewSimpleFile.php(225): OC\Files\SimpleFS\NewSimpleFile->putContent(Resource id #1165)
#33 [internal function]: OC\Files\SimpleFS\NewSimpleFile->OC\Files\SimpleFS\{closure}(Resource id #1165)
#34 /var/www/html/3rdparty/icewind/streams/src/CallbackWrapper.php(113): call_user_func(Object(Closure), Resource id #1165)
#35 /var/www/html/apps/photos/lib/Service/ReverseGeoCoderService.php(139): Icewind\Streams\CallbackWrapper->stream_close()
#36 /var/www/html/apps/photos/lib/Service/ReverseGeoCoderService.php(161): OCA\Photos\Service\ReverseGeoCoderService->buildKDTree()
#37 /var/www/html/apps/photos/lib/Service/ReverseGeoCoderService.php(58): OCA\Photos\Service\ReverseGeoCoderService->loadKdTree()
#38 /var/www/html/apps/photos/lib/Service/MediaPlaceManager.php(74): OCA\Photos\Service\ReverseGeoCoderService->getPlaceForCoordinates(25.058294444444, 121.56169166667)
#39 /var/www/html/apps/photos/lib/Service/MediaPlaceManager.php(40): OCA\Photos\Service\MediaPlaceManager->getPlaceForFile(842261)
#40 /var/www/html/apps/photos/lib/Jobs/AutomaticPlaceMapperJob.php(111): OCA\Photos\Service\MediaPlaceManager->setPlaceForFile(842261)
#41 /var/www/html/apps/photos/lib/Jobs/AutomaticPlaceMapperJob.php(103): OCA\Photos\Jobs\AutomaticPlaceMapperJob->scanFolder(Object(OC\Files\Node\Folder))
#42 /var/www/html/apps/photos/lib/Jobs/AutomaticPlaceMapperJob.php(91): OCA\Photos\Jobs\AutomaticPlaceMapperJob->scanFolder(Object(OC\Files\Node\LazyUserFolder))
#43 /var/www/html/apps/photos/lib/Jobs/AutomaticPlaceMapperJob.php(82): OCA\Photos\Jobs\AutomaticPlaceMapperJob->scanFilesForUser('admin')
#44 /var/www/html/lib/public/BackgroundJob/Job.php(78): OCA\Photos\Jobs\AutomaticPlaceMapperJob->run(NULL)
#45 /var/www/html/lib/public/BackgroundJob/TimedJob.php(103): OCP\BackgroundJob\Job->start(Object(OC\BackgroundJob\JobList))
#46 /var/www/html/lib/public/BackgroundJob/TimedJob.php(93): OCP\BackgroundJob\TimedJob->start(Object(OC\BackgroundJob\JobList))
#47 /var/www/html/cron.php(152): OCP\BackgroundJob\TimedJob->execute(Object(OC\BackgroundJob\JobList), Object(OC\Log))
```

**Comment by socialize-IT** [comment]
Seems related to https://github.com/owncloud/music/issues/1050 .

-------- Ursprüngliche Nachricht --------
Von: JAN ***@***.***>
Datum: 24.03.23 07:36 (GMT+01:00)
An: owncloud/music ***@***.***>
Cc: Steffen Windhab, Comment ***@***.***>
Betreff: Re: [owncloud/music] Please support PHP 8.2 (Issue #1056)


additional log in NC 26:

Error: Class "OCP\AppFramework\Db\Mapper" not found in /var/www/html/custom_apps/music/lib/Db/BaseMapper.php:29
Stack trace:
#0 /var/www/html/lib/composer/composer/ClassLoader.php(578): include()
#1 /var/www/html/lib/composer/composer/ClassLoader.php(432): Composer\Autoload\ClassLoader::Composer\Autoload\{closure}('/var/www/html/c...')
#2 /var/www/html/custom_apps/music/lib/Db/ArtistMapper.php(24): Composer\Autoload\ClassLoader->loadClass('OCA\\Music\\Db\\Ba...')
#3 /var/www/html/lib/composer/composer/ClassLoader.php(578): include('/var/www/html/c...')
#4 /var/www/html/lib/composer/composer/ClassLoader.php(432): Composer\Autoload\ClassLoader::Composer\Autoload\{closure}('/var/www/html/c...')
#5 /var/www/html/custom_apps/music/lib/App/Music.php(363): Composer\Autoload\ClassLoader->loadClass('OCA\\Music\\Db\\Ar...')
#6 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(171): OCA\Music\App\Music->OCA\Music\App\{closure}(Object(OC\AppFramework\DependencyInjection\DIContainer))
#7 /var/www/html/3rdparty/pimple/pimple/src/Pimple/Container.php(122): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(Pimple\Container))
#8 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(138): Pimple\Container->offsetGet('ArtistMapper')
#9 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(487): OC\AppFramework\Utility\SimpleContainer->query('ArtistMapper')
#10 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(465): OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback('ArtistMapper')
#11 /var/www/html/custom_apps/music/lib/App/Music.php(272): OC\AppFramework\DependencyInjection\DIContainer->query('ArtistMapper')
#12 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(171): OCA\Music\App\Music->OCA\Music\App\{closure}(Object(OC\AppFramework\DependencyInjection\DIContainer))
#13 /var/www/html/3rdparty/pimple/pimple/src/Pimple/Container.php(122): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(Pimple\Container))
#14 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(138): Pimple\Container->offsetGet('ArtistBusinessL...')
#15 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(487): OC\AppFramework\Utility\SimpleContainer->query('ArtistBusinessL...')
#16 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(465): OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback('ArtistBusinessL...')
#17 /var/www/html/custom_apps/music/lib/App/Music.php(576): OC\AppFramework\DependencyInjection\DIContainer->query('ArtistBusinessL...')
#18 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(171): OCA\Music\App\Music->OCA\Music\App\{closure}(Object(OC\AppFramework\DependencyInjection\DIContainer))
#19 /var/www/html/3rdparty/pimple/pimple/src/Pimple/Container.php(122): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(Pimple\Container))
#20 /var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php(138): Pimple\Container->offsetGet('Scanner')
#21 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(487): OC\AppFramework\Utility\SimpleContainer->query('Scanner')
#22 /var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php(465): OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback('Scanner')
#23 /var/www/html/custom_apps/music/lib/Hooks/FileHooks.php(75): OC\AppFramework\DependencyInjection\DIContainer->query('Scanner')
#24 /var/www/html/custom_apps/music/lib/Hooks/FileHooks.php(62): OCA\Music\Hooks\FileHooks::handleUpdated(Object(OC\Files\Node\File), Object(OC\AppFramework\DependencyInjection\DIContainer))
#25 [internal function]: OCA\Music\Hooks\FileHooks::updated(Object(OC\Files\Node\File))
#26 /var/www/html/lib/private/Hooks/EmitterTrait.php(105): call_user_func_array(Array, Array)
#27 /var/www/html/lib/private/Hooks/PublicEmitter.php(40): OC\Hooks\BasicEmitter->emit('\\OC\\Files', 'postWrite', Array)
#28 /var/www/html/lib/private/Files/Node/Root.php(143): OC\Hooks\PublicEmitter->emit('\\OC\\Files', 'postWrite', Array)
#29 /var/www/html/lib/private/Files/Node/Node.php(129): OC\Files\Node\Root->emit('\\OC\\Files', 'postWrite', Array)
#30 /var/www/html/lib/private/Files/Node/Folder.php(200): OC\Files\Node\Node->sendHooks(Array, Array)
#31 /var/www/html/lib/private/Files/SimpleFS/NewSimpleFile.php(121): OC\Files\Node\Folder->newFile('cities1000.csv', Resource id #1165)
#32 /var/www/html/lib/private/Files/SimpleFS/NewSimpleFile.php(225): OC\Files\SimpleFS\NewSimpleFile->putContent(Resource id #1165)
#33 [internal function]: OC\Files\SimpleFS\NewSimpleFile->OC\Files\SimpleFS\{closure}(Resource id #1165)
#34 /var/www/html/3rdparty/icewind/streams/src/CallbackWrapper.php(113): call_user_func(Object(Closure), Resource id #1165)
#35 /var/www/html/apps/photos/lib/Service/ReverseGeoCoderService.php(139): Icewind\Streams\CallbackWrapper->stream_close()
#36 /var/www/html/apps/photos/lib/Service/ReverseGeoCoderService.php(161): OCA\Photos\Service\ReverseGeoCoderService->buildKDTree()
#37 /var/www/html/apps/photos/lib/Service/ReverseGeoCoderService.php(58): OCA\Photos\Service\ReverseGeoCoderService->loadKdTree()
#38 /var/www/html/apps/photos/lib/Service/MediaPlaceManager.php(74): OCA\Photos\Service\ReverseGeoCoderService->getPlaceForCoordinates(25.058294444444, 121.56169166667)
#39 /var/www/html/apps/photos/lib/Service/MediaPlaceManager.php(40): OCA\Photos\Service\MediaPlaceManager->getPlaceForFile(842261)
#40 /var/www/html/apps/photos/lib/Jobs/AutomaticPlaceMapperJob.php(111): OCA\Photos\Service\MediaPlaceManager->setPlaceForFile(842261)
#41 /var/www/html/apps/photos/lib/Jobs/AutomaticPlaceMapperJob.php(103): OCA\Photos\Jobs\AutomaticPlaceMapperJob->scanFolder(Object(OC\Files\Node\Folder))
#42 /var/www/html/apps/photos/lib/Jobs/AutomaticPlaceMapperJob.php(91): OCA\Photos\Jobs\AutomaticPlaceMapperJob->scanFolder(Object(OC\Files\Node\LazyUserFolder))
#43 /var/www/html/apps/photos/lib/Jobs/AutomaticPlaceMapperJob.php(82): OCA\Photos\Jobs\AutomaticPlaceMapperJob->scanFilesForUser('admin')
#44 /var/www/html/lib/public/BackgroundJob/Job.php(78): OCA\Photos\Jobs\AutomaticPlaceMapperJob->run(NULL)
#45 /var/www/html/lib/public/BackgroundJob/TimedJob.php(103): OCP\BackgroundJob\Job->start(Object(OC\BackgroundJob\JobList))
#46 /var/www/html/lib/public/BackgroundJob/TimedJob.php(93): OCP\BackgroundJob\TimedJob->start(Object(OC\BackgroundJob\JobList))
#47 /var/www/html/cron.php(152): OCP\BackgroundJob\TimedJob->execute(Object(OC\BackgroundJob\JobList), Object(OC\Log))

**Comment by datenleck** [comment]
"Message": "Class \"OCP\\AppFramework\\Db\\Mapper\" not found",

No, that app is not working if I change the maximum PHP version from 8.1 to 8.2 in the file  appinfo/info.xml

**Comment by JoshuaPettus** [comment]
Mapper being decommissioned by NC is a separate issue independent of php 8.2  #1050

**Comment by amiga23** [comment]
Also the nextcloud upgrade process stops with error when upgrading from 25 to 26 and music app was already installed.
```
Error: Class "OCP\AppFramework\Db\Mapper" not found
```

**Comment by amiga23** [comment]
A PR that fixes the issue was already merged, waiting for release ...
https://github.com/owncloud/music/pull/1055

**Comment by paulijar** [comment]
The support for PHP 8.2 (as well as for NC 26) has now been released in Music v1.8.2.
